### PR TITLE
Add Extensions page, update What's New page, change how extensions are installed

### LINKED ISSUE #676: Marketplace & Library for extensions
### Suggested new feature or improvement

Dev Home is built to be extensible.  Users should have an easy way to find, install, & manage extensions built for Dev Home. This includes an **Extensions Marketplace** page & an **Extensions Library** page to support accessing & managing first- and third-party-built extensions directly within Dev Home.

**Extensions Marketplace:**
- Dev Home provides UI for users to discover & browse all of the Dev Home extensions published in the Microsoft Store
- Users can install extensions directly from Dev Home

**Extensions Library:**
- Dev Home provides UI for users to view all installed Dev Home extensions
- Users can check for updates, update, uninstall, sort, & view high-level details (i.e. version) for their installed Dev Home extensions

### Scenario

Getting / finding new Dev Home extensions.  

Managing installed Dev Home extensions.

### Additional details

_No response_

### PR DESCRIPTION
**Pull request by krschau**
## Summary of the pull request

This is quite a large pull request, but all changes have been reviewed in smaller PRs that can be found in the list of commits below.

This change adds an "Extension Library" where users can install Dev Home extensions from the store. For now, there is only one extension, so we have a static file with the list that we can edit as the extensions grow, and eventually query directly from the store.

This also adds Extensions the "What's new" page.

We also no longer install the Dev Home GitHub Extension automatically.

image
image

## PR checklist
- [x] Closes #676
- [ ] Tests added/passed
- [ ] Documentation updated

### CODE REVIEW

File: tools/ExtensionLibrary/DevHome.ExtensionLibrary/Assets/extensionResult.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	  {
+        "PackageFamilyName": "Microsoft.Windows.DevHomeGitHubExtension_8wekyb3d8bbwe"
+      }
+    },
+    {
+      "LocalizedProperties": 
+	  [
+        {
+          "PublisherName": "Microsoft Corporation",
+          "ProductTitle": "Dev Home (Preview)"
```
Comment (EricJohnson327) [inline tools/ExtensionLibrary/DevHome.ExtensionLibrary/Assets/extensionResult.json:24]: Why are we adding Dev Home (Preview) as an extension?

**Review by EricJohnson327 — APPROVED** [review]

File: tools/ExtensionLibrary/DevHome.ExtensionLibrary/Assets/extensionResult.json (hunk 1 above)
Comment (EricJohnson327) [reply tools/ExtensionLibrary/DevHome.ExtensionLibrary/Assets/extensionResult.json:24]: Never mind, I see we're ignoring it later on.  This is just to mimic what the store will return to us in the future.  Perfect!

File: src/Styles/Thickness.xaml [hunk 2]
```diff
@@ -38,4 +38,6 @@
 
     <Thickness x:Key="SettingsCardMargin">0,0,0,4</Thickness>
     <x:Double x:Key="SettingsCardSpacing">4</x:Double>
+
```
Comment (dhoehna) [inline src/Styles/Thickness.xaml:41]: NIT: Remove extra line.

File: src/ViewModels/WhatsNewViewModel.cs [hunk 3]
```diff
@@ -10,15 +10,105 @@ namespace DevHome.ViewModels;
...
+    {
+        BigSource.Add(card);
+    }
+
+    // When the width is too small, the big cards should go into the normal sized cards collection.
+    // To do this, we first merge all the cards together in a List, then we sort them by priority,
+    // and finally throw them back in the normal sized cards collection.
+    public void SwitchToSmallerView()
+    {
+        if (BigSource.Count == 0)
```
Comment (dhoehna) [inline src/ViewModels/WhatsNewViewModel.cs:36]: NIT: !BigSource.Any()

File: src/Styles/Thickness.xaml (hunk 2 above)
Comment (EricJohnson327) [reply src/Styles/Thickness.xaml:41]: The extra line is expected because it's for a different page.

File: src/Views/WhatsNewPage.xaml.cs [hunk 4]
```diff
@@ -61,6 +61,25 @@ private async void OnLoaded(object sender, Microsoft.UI.Xaml.RoutedEventArgs e)
 
             ViewModel.AddCard(card);
         }
+
+        var whatsNewBigCards = BigFeaturesContainer.Resources
+            .Where((item) => item.Value.GetType() == typeof(WhatsNewCard))
```
Comment (dhoehna) [inline src/Views/WhatsNewPage.xaml.cs:66]: NIT: Extra () around item.

**Review by dhoehna — APPROVED** [review]

File: tools/ExtensionLibrary/DevHome.ExtensionLibrary/ViewModels/InstalledPackageViewModel.cs [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+            var localSettingsService = Application.Current.GetService<ILocalSettingsService>();
+            return localSettingsService.ReadSettingAsync<bool>(ExtensionUniqueId + "-ExtensionDisabled");
+        }).Result;
+        return !isDisabled;
+    }
+
+    [RelayCommand]
+    private void NavigateSettings()
+    {
+        // TODO: nothing implements ISettingsProvider yet, so this will not be called
```
Comment (vineeththomasalex) [inline tools/ExtensionLibrary/DevHome.ExtensionLibrary/ViewModels/InstalledPackageViewModel.cs:64]: Might have benefited from a Throw NotImplemented, but not a necessary change.

**Review by vineeththomasalex — APPROVED** [review]

File: tools/ExtensionLibrary/DevHome.ExtensionLibrary/ViewModels/InstalledPackageViewModel.cs (hunk 5 above)
Comment (krschau) [reply tools/ExtensionLibrary/DevHome.ExtensionLibrary/ViewModels/InstalledPackageViewModel.cs:64]: Will be implemented tomorrow.

**Pull request merged**